jetbrains-core/src/software/aws/toolkits/jetbrains/services/s3/S3TreeCellRenderer.kt

```diff
@@ -0,0 +1,37 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+package software.aws.toolkits.jetbrains.services.s3
+
+import software.aws.toolkits.jetbrains.services.s3.bucketEditor.S3KeyNode
+import java.awt.Component
+import javax.swing.JLabel
+import javax.swing.JTree
+import javax.swing.tree.DefaultMutableTreeNode
+import javax.swing.tree.DefaultTreeCellRenderer
+
+class S3TreeCellRenderer : DefaultTreeCellRenderer() {
+
+    override fun getTreeCellRendererComponent(
+        tree: JTree?,
+        value: Any?,
+        sel: Boolean,
+        expanded: Boolean,
+        leaf: Boolean,
+        row: Int,
+        hasFocus: Boolean
+    ): Component {
+        val renderer = super.getTreeCellRendererComponent(tree, value, sel, expanded, leaf, row, hasFocus) as JLabel
+        if (value is DefaultMutableTreeNode) {
+            val userObject = value.userObject
+            if (userObject is S3KeyNode) {
```

Comment:
I thought we had different representations of 'object' vs 'directory' as a base concept (extensions of `S3KeyNode`). If so this (and the nested if) could be replaced with a `when` block.